src/variants/lambda/implementation.mligo

```diff
@@ -0,0 +1,101 @@
+// SPDX-FileCopyrightText: 2022 Tezos Commons
+// SPDX-License-Identifier: LicenseRef-MIT-TC
+
+#if !VARIANT
+#define VARIANT
+
+#include "error_codes.mligo"
+#include "helper/unpack.mligo"
+#include "variants/lambda/types.mligo"
+#include "types.mligo"
+
+// -----------------------------------------------------------------
+// Proposal check
+//
+// This should implement the proposal_check logic of the variant and should
+// verify whether a proposal can be submitted.  It checks 2 things: the
+// proposal itself and the amount of tokens frozen upon submission.  It allows
+// the DAO to reject a proposal by arbitrary logic and captures bond
+// requirements
+// -----------------------------------------------------------------
+
+let handler_exists(handler_name, ce : string * contract_extra) : bool =
+  Big_map.mem handler_name ce.lambdas
+
+let unpack_proposal_metadata (pm: proposal_metadata) : lambda_dao_proposal_metadata =
+  match ((Bytes.unpack pm) : (lambda_dao_proposal_metadata option)) with
+  | Some (v) -> v
+  | None -> (failwith unpacking_proposal_metadata_failed : lambda_dao_proposal_metadata)
+
+let proposal_check (propose_params, ce : propose_params * contract_extra) : unit =
+  match unpack_proposal_metadata(propose_params.proposal_metadata) with
+    | Add_handler ahp -> if handler_exists(ahp.name, ce) then (failwith proposal_handler_exists : unit) else unit
+    | Remove_handler rhp -> if handler_exists(rhp, ce) then unit else (failwith proposal_handler_not_found : unit)
```

Comment:
The `remove_handler` proposal should only succeed if the handler is active, so we should ensure if the handler is active as well.

Reply:
Oh, about this, I think as per spec, we planned `remove_handler` to do nothing if the handler is already de-activated, so I think this validation is not required.

Reply:
Hmm I think we do nothing when in `decision_lambda`, but in `proposal_check` it mentions that it check for `active` handler:

![image](https://user-images.githubusercontent.com/8400953/180253515-e1b0b58a-97cf-4f0a-a608-6ff9b43be6d9.png)

So I think the validation is still required in this case.

Reply:
You are right. I got confused there for a minute.

Reply:
Have added this and a test.